include/picongpu/particles/Synchrotron/AlgorithmSynchrotron.hpp

```diff
@@ -0,0 +1,428 @@
+                HDINLINE float_X // return type -> energy of a photon relative to the electron energy
+                operator()(
+                    const T_Worker& worker,
+                    const BType bField,
+                    const EType eField,
+                    ParticleType& parentElectron,
+                    float_64 randNr1,
+                    float_64 randNr2,
+                    GridBuffer<float_X, 2>::DataBoxType F1F2DeviceBuff,
+                    GridBuffer<int32_t, 1>::DataBoxType failedRequirementQBuff) const
+                {
+                    constexpr int16_t minZqExponent = params::InterpolationParams::minZqExponent;
+                    constexpr int16_t maxZqExponent = params::InterpolationParams::maxZqExponent;
+                    constexpr float_X interpolationPoints
+                        = static_cast<float_X>(params::InterpolationParams::numberTableEntries - 1);
+                    constexpr float_X stepWidthLogatihmicScale
+                        = (static_cast<float_X>(maxZqExponent - minZqExponent) / interpolationPoints);
+
+                    /// zq = 2/3 * chi^(-1) * delta / (1 - delta) ;
+                    ///     chi - ratio of the typical photon energy to the electron kinetic energy
+                    ///     delta - the ratio of the photon energy to the electron energy
+                    /// see Extended particle-in-cell schemes for physics in ultrastrong laser fields: Review and
+                    /// developments, A. Gonoskov et.Al.
+
+                    // delta
+                    float_X const r1r1 = randNr1 * randNr1;
+                    float_X const delta = r1r1 * randNr1;
+
+                    // calculate Heff
+                    float_X const mass = attribute::getMass(parentElectron[weighting_], parentElectron);
+                    float3_X const vel = Velocity()(parentElectron[momentum_], mass);
+
+                    float_X const Vmag = pmacc::math::l2norm(vel);
+                    float3_X const crossVB = pmacc::math::cross(vel, bField);
+                    float3_X const Vnorm = vel / Vmag; // Velocity normalized -> not L2Norm
+                    float_X const dotVnormE = pmacc::math::dot(Vnorm, eField);
+                    float3_X const eFieldPlusCrossVB = eField + crossVB;
+
+                    float_X HeffValue = pmacc::math::dot(eFieldPlusCrossVB, eFieldPlusCrossVB) - dotVnormE * dotVnormE;
+                    if(HeffValue <= 0)
+                    {
+                        return 0;
+                    }
+                    HeffValue = math::sqrt(HeffValue);
+
+
+                    // Calculate chi
+                    float_X const gamma = Gamma()(parentElectron[momentum_], mass);
+                    float_X const chi = -ELECTRON_CHARGE * HBAR * HeffValue * gamma
+                        / (ELECTRON_MASS * ELECTRON_MASS * SPEED_OF_LIGHT * SPEED_OF_LIGHT * SPEED_OF_LIGHT);
```

Comment:
Could you write this with a const variable of the Schwinger limit to enhance readability and ease a quick understanding.
```suggestion
                    float_X const E_Schwinger = ELECTRON_MASS * ELECTRON_MASS * SPEED_OF_LIGHT * SPEED_OF_LIGHT * SPEED_OF_LIGHT / (-ELECTRON_CHARGE * HBAR)
                    float_X const chi =  HeffValue * gamma / E_Schwinger;
```

Reply:
I think I made it that way in the past to divide only once.

Reply:
Made it like this:
`
float_X const inverse_E_Schwinger = (-ELECTRON_CHARGE * HBAR) / ELECTRON_MASS * ELECTRON_MASS * SPEED_OF_LIGHT * SPEED_OF_LIGHT * SPEED_OF_LIGHT;
float_X const chi = HeffValue * gamma * inverse_E_Schwinger;`